kDriveCore/Utils/FileImportHelper.swift

```diff
@@ -223,20 +240,76 @@ public class FileImportHelper {
-                        DDLogError("[FileImportHelper] Error while getting file: \(error)")
+                    case .failure(let error):
+                        DDLogError("[FileImportHelper] Error while getting imageData file: \(error)")
                         errorCount += 1
                     }
                     dispatchGroup.leave()
                 }
                 progress.addChild(childProgress, withPendingUnitCount: perItemUnitCount)
-            } else {
+            case .isCompressedData(let typeIdentifier):
+                let childProgress = getFile(from: itemProvider, typeIdentifier: typeIdentifier) { result in
+                    switch result {
+                    case .success((let filename, let fileURL)):
+                        items.append(ImportedFile(name: filename, path: fileURL, uti: UTI(typeIdentifier) ?? .data))
+                    case .failure(let error):
+                        DDLogError("[FileImportHelper] Error while getting compressedData file: \(error)")
+                        errorCount += 1
+                    }
+                    dispatchGroup.leave()
+                }
+                progress.addChild(childProgress, withPendingUnitCount: perItemUnitCount)
+            case .isDirectory:
+
+                // TODO: use itemProvider.zippedRepresentation
+
+                let tmpDirectoryURL = pathProvider.tmpDirectoryURL
+                let tempURL = tmpDirectoryURL.appendingPathComponent("\(UUID().uuidString).zip")
+
+                _ = itemProvider.loadObject(ofClass: URL.self) { path, error in
+                    guard error == nil, let path: URL = path else {
+                        progress.completedUnitCount += perItemUnitCount
+                        errorCount += 1
+                        dispatchGroup.leave()
+                        return
+                    }
+
+                    // compress content of folder and move it somewhere we can safely store it for upload
+                    var error: NSError?
+                    coordinator.coordinate(readingItemAt: path, options: [.forUploading], error: &error) { zipURL in
+                        do {
+                            let zipName = zipURL.lastPathComponent
+                            try fileManager.moveItem(at: zipURL, to: tempURL)
+
+                            items.append(ImportedFile(name: zipName, path: tempURL, uti: UTI.zip))
+                            progress.completedUnitCount += perItemUnitCount
+                            dispatchGroup.leave()
+                        } catch {
+                            DDLogError("could not move zipped file:\(error)")
+                            progress.completedUnitCount += perItemUnitCount
+                            errorCount += 1
+                            dispatchGroup.leave()
+                            return
+                        }
+                    }
+                }
+
+            case .isMiscellaneous, .none:
```

Comment:
I have an issue with arbitrary types, that should be handeled as a `file`. Working on a fix.

Reply:
fix